packaging_assets/package.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+# Copyright 2022 Contributors to the Parsec project.
+# SPDX-License-Identifier: Apache-2.0
+
+# Create a quickstart package
+
+ARCH="x86"
+OS="linux"
+PACKAGE_PATH=$(pwd)
+WORK_DIR="/tmp/workdir-parsec/"
+JOBS_NUMBER=1
+SCRIPT_PATH=$(realpath $0)
+ASSETS_DIR=$(dirname $SCRIPT_PATH)
+PARSEC_DIR=$(dirname $ASSETS_DIR)
+VERSION=$(cargo metadata --format-version=1 --no-deps --offline | jq -r '.packages[0].version')
+
+# Usage
+USAGE_STR=\
+"Usage:\n"\
+"package.sh [Options]\n"\
+"Options:\n"\
+" -o {path}: Output absolute path, the default path is the current directory i.e. $(pwd)\n"\
+" -j {jobs}: Number of parallel jobs, Default is $JOBS_NUMBER"\
+" -h : Display this help menu\n"
+
+# Flags
+while getopts v:o:j:h flag
+do
+    case "${flag}" in
+        o) PACKAGE_PATH=${OPTARG};;
+        j) JOBS_NUMBER=${OPTARG};;
+        h) echo -e $USAGE_STR; exit 0;;
+    esac
+done
+
+check_version() {
+    echo "Checking version"
+    if [ -z "$VERSION" ];then
+            echo "Couldn't extract the version!" >&2
+            exit 1
+    fi
+}
+
+check_release_tag() {
+    CURRENT_TAG=$(git name-rev --tags HEAD | cut -d "/" -f 2)
+    LATTEST_TAG=$(git tag --sort=committerdate | tail -1)
+    if [ -z "$LATTEST_TAG" ];then
+        echo "Warning:No tags"
+    fi
+    if [ "$LATTEST_TAG" == "$CURRENT_TAG" ]; then
+        echo "Packaging release tag: $LATTEST_TAG"
+    else
+        echo "Warning: The current HEAD does't match the latest tagged"
+        echo "Warning: Please checkout the latest tag : $LATTEST_TAG"
+        read  -n 1 -p "Do you want to continue anyway [y/n]?" choice
+        if [ "$choice" != "y" ]; then
+            exit 1
+        fi
+    fi
+}
+
+cleanup()
+{
+    echo "Clean up"
+    rm -rf $WORK_DIR
+}
+
+pre_package() {
+    # Construct package name
+    PACKAGE_DIR=quickstart-$VERSION-${OS}_$ARCH
+
+    # Create a temp work directory for parsec service
+    mkdir -p $WORK_DIR/parsec
+
+    # Create the package directory
+    mkdir $WORK_DIR/$PACKAGE_DIR
+}
+
+build_parsec_service() {
+    # Package Parsec
+    echo "Building Parsec"
+
+    CARGO_TARGET_DIR=$WORK_DIR/parsec/ cargo build -j $JOBS_NUMBER --release --features mbed-crypto-provider --manifest-path $PARSEC_DIR/Cargo.toml
+}
+
+build_parsec_tool() {
+    # Package Parsec-tool
+    echo "Building Parsec-tool"
+
+    git clone https://github.com/parallaxsecond/parsec-tool $WORK_DIR/parsec-tool
```

Comment:
do we want to check out the latest tag for `parsec-tool` demo as well?

Reply:
Good point, I think we need to agree on how parsec and parsec-tool synced together.
Taking the latest release makes sense, but when they are not synced, which version should we pick?
Should it be (the version) provided in case the latest release of the parsec-tool is not synced?

Reply:
As discussed, Latest tag of parsec-tool will be used